runtime/runtime_impl_jetty9/src/main/java/com/google/apphosting/runtime/jetty9/JettyRequestAPIData.java

```diff
@@ -329,6 +327,41 @@ public String getScheme() {
           public boolean isSecure() {
             return isSecure;
           }
+
+          @Override
+          public String getRemoteAddr() {
+            return finalUserIp;
```

Comment:
[nit] Why not just use `userIp` directly. It doesn't mutate between these lines right?

Reply:
Not possible because it is reassigned above and the variable has to be "effectively final", the compiler will give the error:
> Variable 'userIp' is accessed from within inner class, needs to be final or effectively final